packages/editor/src/store/controls.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * WordPress dependencies
+ */
+import { default as triggerFetch } from '@wordpress/api-fetch';
+import { createRegistryControl } from '@wordpress/data';
+
+export function apiFetch( request ) {
+	return {
+		type: 'API_FETCH',
+		request,
+	};
+}
+
+export function select( reducerKey, selectorName, ...args ) {
+	return {
+		type: 'SELECT',
+		reducerKey,
+		selectorName,
+		args,
+	};
+}
+
+export function resolveSelect( reducerKey, selectorName, ...args ) {
+	return {
+		type: 'RESOLVE_SELECT',
+		reducerKey,
+		selectorName,
+		args,
+	};
+}
+
+export function dispatch( reducerKey, actionName, ...args ) {
+	return {
+		type: 'DISPATCH',
+		reducerKey,
+		actionName,
+		args,
+	};
+}
+
+export default {
+	API_FETCH( { request } ) {
+		return triggerFetch( request );
+	},
+	SELECT: createRegistryControl(
+		( registry ) => ( { reducerKey, selectorName, args } ) => {
+			return registry.select( reducerKey )[ selectorName ]( ...args );
+		}
+	),
+	DISPATCH: createRegistryControl(
+		( registry ) => ( { reducerKey, actionName, args } ) => {
+			return registry.dispatch( reducerKey )[ actionName ]( ...args );
+		}
+	),
+	RESOLVE_SELECT: createRegistryControl(
```

Comment:
It'd be nice if the registry exposed some sort of helper for determining if a selector has a resolver (i.e. `registry.getStore( reducerKey ).hasResolverFor( selectorName )`).  That way this code could be condensed to just having a `select` control that conditionally resolves if there is a resolver.

Reply:
Agreed, that some help from the registry here would be good, this could be addressed in a separate PR maybe?